riptide-failsafe/src/main/java/org/zalando/riptide/failsafe/RetryException.java

```diff
@@ -0,0 +1,18 @@
+package org.zalando.riptide.failsafe;
+
+import org.springframework.http.client.ClientHttpResponse;
+
+import java.io.IOException;
+
+final class RetryException extends RuntimeException {
+
+    RetryException(final ClientHttpResponse response) throws IOException {
+        super(formatMessage(response));
+    }
+
+    private static String formatMessage(final ClientHttpResponse response) throws IOException {
```

Comment:
I copied this over from the `NoRouteException` and stripped the read-some-bytes part. It might make sense to generalize this and use the same implementation in both exceptions, IMO this would make sense.